<launch>	
	<!-- Platform executors -->
	<group ns="fmExecutors">
		<node pkg="position_action_server" name="position_planner" type="positionGoalActionServer.py" output="screen">
			<param name="cmd_vel_topic" value="/fmSignals/cmd_vel" />
			<param name="odom_topic" value="fmKnowledge/odom" />
			<param name="max_linear_velocity" value="1"/>
			<param name="max_angular_velocity" value="4.0"/>
			<param name="straight_line" value="True"/>
			<param name="straight_line_angle_error" value="0.5"/>
		</node>
	</group>
	
	<!-- Platform Controllers -->
	<group ns="fmControllers">
		<node pkg="differential_ifk" name="differential_driver" type="differential_ifk_node" output="screen" >
			<param name="hl_subscriber_topic" value="/fmSignals/cmd_vel"/>
			<param name="ll_publisher_topic_left" value="/fmSignals/cmd_vel_left"/>
			<param name="ll_publisher_topic_right" value="/fmSignals/cmd_vel_right"/>
		</node>
		
		<node pkg="sdu_pichi" name="robot_interface" type="sdu_pichi_node" output="screen" >
			<param name="cmd_vel_ch1_topic" value="/fmSignals/cmd_vel_left"/>
			<param name="cmd_vel_ch2_topic" value="/fmSignals/cmd_vel_right"/>
			<param name="serial_rx_topic" value="/fmData/robot_rx"/>
			<param name="serial_tx_topic" value="/fmData/robot_tx"/>
			<param name="deadman_topic" value="/fmSafety/deadman"/>
			<param name="encoder_ch1_topic" value="/fmInformation/encoder_left"/>
			<param name="encoder_ch2_topic" value="/fmInformation/encoder_right"/>	
			<param name="power_ch1_topic" value="/fmData/power_left"/>
			<param name="power_ch2_topic" value="/fmData/power_right"/>
			<param name="status_topic" value="/fmData/status"/>	
			<param name="mps_to_rpm" value="1432"/>        
        </node>
	</group>
</launch>